In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 5GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/breast-cancer-wisconsin-data/data.csv


In [7]:
data=pd.read_csv('/kaggle/input/breast-cancer-wisconsin-data/data.csv')

In [8]:
data.head(10)

id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   
5    843786         M        12.45         15.70           82.57      477.1   
6    844359         M        18.25         19.98          119.60     1040.0   
7  84458202         M        13.71         20.83           90.20      577.9   
8    844981         M        13.00         21.82           87.50      519.8   
9  84501001         M        12.46         24.04           83.97      475.9   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760         0.30010              0.14710   
1          0.08474           0.07864         0.08690              0.07017   
2          0.10960           0.15990         0.19740              0.12790   
3          0.14250           0.28390         0.24140              0.10520   
4          0.10030           0.13280         0.19800              0.10430   
5          0.12780           0.17000         0.15780              0.08089   
6          0.09463           0.10900         0.11270              0.07400   
7          0.11890           0.16450         0.09366              0.05985   
8          0.12730           0.19320         0.18590              0.09353   
9          0.11860           0.23960         0.22730              0.08543   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0  ...          17.33           184.60      2019.0            0.1622   
1  ...          23.41           158.80      1956.0            0.1238   
2  ...          25.53           152.50      1709.0            0.1444   
3  ...          26.50            98.87       567.7            0.2098   
4  ...          16.67           152.20      1575.0            0.1374   
5  ...          23.75           103.40       741.6            0.1791   
6  ...          27.66           153.20      1606.0            0.1442   
7  ...          28.14           110.60       897.0            0.1654   
8  ...          30.73           106.20       739.3            0.1703   
9  ...          40.68            97.65       711.4            0.1853   

   compactness_worst  concavity_worst  concave points_worst  symmetry_worst  \
0             0.6656           0.7119                0.2654          0.4601   
1             0.1866           0.2416                0.1860          0.2750   
2             0.4245           0.4504                0.2430          0.3613   
3             0.8663           0.6869                0.2575          0.6638   
4             0.2050           0.4000                0.1625          0.2364   
5             0.5249           0.5355                0.1741          0.3985   
6             0.2576           0.3784                0.1932          0.3063   
7             0.3682           0.2678                0.1556          0.3196   
8             0.5401           0.5390                0.2060          0.4378   
9             1.0580           1.1050                0.2210          0.4366   

   fractal_dimension_worst  Unnamed: 32  
0                  0.11890          NaN  
1                  0.08902          NaN  
2                  0.08758          NaN  
3                  0.17300          NaN  
4                  0.07678          NaN  
5                  0.12440          NaN  
6                  0.08368          NaN  
7                  0.11510          NaN  
8                  0.10720          NaN  
9                  0.20750          NaN  

[10 rows x 33 columns]

In [9]:
X=data.iloc[:,2:32].values

y=data.iloc[:,1].values

In [10]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2)

In [11]:
from sklearn.svm import SVC
rbf_classifier = SVC(kernel="rbf")

In [12]:
rbf_classifier.fit(X_train, y_train)


SVC()

In [13]:
y_pred = rbf_classifier.predict(X_test)

In [14]:
from sklearn.metrics import accuracy_score
accuracy_score(y_test, y_pred)

0.9122807017543859

In [15]:
poly_classifier = SVC(kernel="poly",degree=2)

In [16]:
poly_classifier.fit(X_train, y_train)

SVC(degree=2, kernel='poly')

In [19]:
y_pred_poly = poly_classifier.predict(X_test)

In [20]:
accuracy_score(y_test, y_pred_poly)

0.9122807017543859

In [31]:
poly_classifier = SVC(kernel="poly",degree=10)
poly_classifier.fit(X_train, y_train)
y_pred_poly = poly_classifier.predict(X_test)
accuracy_score(y_test, y_pred_poly)

0.9122807017543859

In [32]:
param_dist={
    "kernel":["linear","rbf","poly"],
    "degree":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,None]
}

In [33]:
from sklearn.model_selection import GridSearchCV
grid=GridSearchCV(rbf_classifier,param_grid=param_dist,cv=10,n_jobs=-1)

In [34]:
grid.fit(X_train,y_train)

GridSearchCV(cv=10, estimator=SVC(), n_jobs=-1,
             param_grid={'degree': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
                                    14, 15, 16, 17, 18, 19, 20, None],
                         'kernel': ['linear', 'rbf', 'poly']})

In [35]:
grid.best_estimator_

SVC(degree=1, kernel='linear')

In [36]:
grid.best_score_

0.9603381642512078

In [37]:
grid_2=GridSearchCV(poly_classifier,param_grid=param_dist,cv=10,n_jobs=-1)

In [38]:
grid_2.fit(X_train,y_train)

GridSearchCV(cv=10, estimator=SVC(degree=10, kernel='poly'), n_jobs=-1,
             param_grid={'degree': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
                                    14, 15, 16, 17, 18, 19, 20, None],
                         'kernel': ['linear', 'rbf', 'poly']})

In [39]:
grid_2.best_estimator_

SVC(degree=1, kernel='linear')

In [40]:
grid_2.best_score_

0.9603381642512078

# Hence it is seen that the best kernel  to perform  SVM  classification on the Breast cancer dataset is Linear and with a accuracy score of 96%.